src/sparseml/transformers/sparsification/sparse_model.py

```diff
@@ -144,6 +141,8 @@ class SparseAutoModel:
     Factory class for creating sparse models using transformers AutoModel classes
     """
 
+    from sparseml.modifiers.quantization.modification import modify_model
+
```

Comment:
Why do we need this here if modification is happening within the quantization modifier?

Reply:
We'd need it for everything but `SparseAutoModelForCausalLM`. Remember that the quantization modifier is only relevant for LMMs, not "classical" transformer models.